src/store/Controller.cc

```diff
@@ -677,20 +677,36 @@ Store::Controller::handleIdleEntry(StoreEntry &e)
 }
 
 void
-Store::Controller::updateOnNotModified(StoreEntry *old, const StoreEntry &newer)
+Store::Controller::updateOnNotModified(StoreEntry *old, StoreEntry &e304)
```

Comment:
I would convert the first parameter to reference type, since the only caller (clientReplyContext::handleIMSReply()) definitely has the old entry:
```suggestion
Store::Controller::updateOnNotModified(StoreEntry &old, StoreEntry &e304)
```

Reply:
I think I did not want to touch the caller. The change you describe is a good one, but feels a bit out of scope here. Let's not do it unless Amos suggests it as well.